Домашнее задание 7

KrisM1307 opened the pull request.

KrisM1307 committed:
Домашнее задание 7

mdubik777 commented on script.js:
```diff
@@ -1,86 +1,132 @@
+    });
+}
+
+makeCard();
```
Кристина,  лучше тут передать параметром items.  Тогда функция будет более универсальная и сможет работать и с другими массива,  а не только с items

makeCard(items);

mdubik777 commented on script.js:
```diff
@@ -1,86 +1,132 @@
+const itemTemplate = document.querySelector('#item-template');
+const shopItems = document.querySelector('#shop-items');
+
+function makeCard() {
```
Тут функция примет параметр ,  например _items  и тогда работать с ним

mdubik777 commented on script.js:
```diff
@@ -1,86 +1,132 @@
+const nothingFound = document.querySelector('#nothing-found');
+
+searchButton.addEventListener('click', function() {
+    const userText = sendInput.value.trim().toLowerCase();
```
А откуда взялся sendInput?)Из-за этого ошибка в консоли и поиск не работает

KrisM1307 replied:
Ой, это я по примеру того, как чат делали смотрела и перепутала с переменной, которая там) но сейчас исправила на правильное и всё равно не работает( при поиске чего-либо просто все карточки пропадают и всё

KrisM1307 committed:
Исправила ошибки

Поиск все равно не работает(

mdubik777 commented on script.js:
```diff
@@ -1,86 +1,132 @@
+
+    shopItems.innerHTML = '';
+
+    if (items.title.includes(userText)) {
```
Кристина,  когда открываешь проект в браузере - всегда открывай консоль. Там бы увидела ошибку)

items - это массив. в этой проверке ты пытаешься взять title у массива,  его естественно нет и падает ошибка

Эту проверку убирай. Сразу фильтруй массив

mdubik777 commented on script.js:
```diff
@@ -1,86 +1,132 @@
+
+    if (items.title.includes(userText)) {
+        const filteredItems = items.filter(function(item) {
+            return item.textContent.toLowerCase().includes(userText);
```
Item -  это элемент массива,  у него нет textContent  и есть title, в котором мы и должны искать то что ввел пользователь. Поэтому будет

item.title.toLowerCase().includes(userText);

mdubik777 commented on script.js:
```diff
@@ -1,86 +1,132 @@
+    if (items.title.includes(userText)) {
+        const filteredItems = items.filter(function(item) {
+            return item.textContent.toLowerCase().includes(userText);
+        });
```
Когда отфильтровала массив,  то нужно проверить его длину. Если она 0 -  значит выводим ничего не найдено,  иначе отрисовываем элементы (след коммент как отрисовать элементы)

KrisM1307 committed:
Исправления

Ничего не найдено теперь получается вывести, а элементы при поиске так и не выводятся, не понимаю, если честно(

mdubik777 commented on script.js:
```diff
@@ -1,86 +1,140 @@
+
+    shopItems.innerHTML = '';
+
+    if (items.includes(userText)) {
```
Кристина,  теперь ты пытаешься проверить включена ли строка,  которую ввел пользователь, в массив. Я написала,  что эта проверка не нужна. Сразу фильтруй массив

mdubik777 commented on script.js:
```diff
@@ -1,86 +1,140 @@
+            return item.title.toLowerCase().includes(userText);
+        });
+
+        shopItems.append(filteredItems);
```
Еще разок. filteredItems  -  это массив. массив нельзя вставить в контейнер,  можно вставить заполненную карточку , а для того чтобы карточку создать для каждого элемента  -  нужно вызвать функцию makeCard и передать туда отфильтрованный массив

mdubik777 commented on script.js:
```diff
@@ -1,86 +1,140 @@
+
+    if (filteredItems.length > 0) {
+        makeCard(items);
+        shopItems.append(filteredItems);
```
filteredItems - это массив. массив нельзя вставить в контейнер. Эту строку 135 удалить

mdubik777 commented on script.js:
```diff
@@ -1,86 +1,140 @@
+    } else {
+        shopItems.innerHTML = '';
+        nothingFound.textContent = 'Ничего не найдено';
+    };
```
Строки 122 -  131 удалить и на их месте просто отфильтровать массив

KrisM1307 committed:
Вроде получилось..
